lib/puppet/provider/templates/webadministration/getapps.ps1.erb

```diff
@@ -1,23 +1,47 @@
 Get-WebApplication | % {
   $name = [string]$_.Path.replace("/","")
   $site = [string]$_.ItemXPath.split("'")[1]
+  $id   = [string]$_.ItemXPath.split("'")[3]
+
+  $user_name = [String](Get-WebConfigurationProperty -Filter "/system.applicationHost/sites/site[@name='${site}' and @id='${id}']/application[@path='/${name}']/virtualDirectory[@path='/']" -Name userName).Value
+  $password  = [String](Get-WebConfigurationProperty -Filter "/system.applicationHost/sites/site[@name='${site}' and @id='${id}']/application[@path='/${name}']/virtualDirectory[@path='/']" -Name password).Value
 
   $sslFlags = @()
-  $sslFlags_raw = [String](Get-WebConfiguration -Location "${site}/${name}" -Filter "system.webserver/security/access").sslFlags
-  if ($sslFlags_raw -ne '') { $sslFlags = $sslFlags_raw -split ',' }
+  $auth_anonymous                   = 'unknown'
+  $auth_basic                       = 'unknown'
+  $auth_clientCertificateMapping    = 'unknown'
+  $auth_iisClientCertificateMapping = 'unknown'
+  $auth_windows                     = 'unknown'
+  $provider_error                   = ''
+
+  # 'Get-WebConfiguration -Location' can fail if the identity cannot access physicalpath.
+  try {
+    $sslFlags_raw                     = [String](Get-WebConfiguration -Location "${site}/${name}" -Filter "system.webserver/security/access").sslFlags
+    if ($sslFlags_raw -ne '') { $sslFlags = $sslFlags_raw -split ',' }
+    $auth_anonymous                   = [bool](Get-WebConfiguration -Location "${site}/${name}" -Filter "system.webserver/security/authentication/anonymousAuthentication").enabled
+    $auth_basic                       = [bool](Get-WebConfiguration -Location "${site}/${name}" -Filter "system.webserver/security/authentication/basicAuthentication").enabled
+    $auth_clientCertificateMapping    = [bool](Get-WebConfiguration -Location "${site}/${name}" -Filter "system.webserver/security/authentication/clientCertificateMappingAuthentication").enabled
+    $auth_iisClientCertificateMapping = [bool](Get-WebConfiguration -Location "${site}/${name}" -Filter "system.webserver/security/authentication/iisClientCertificateMappingAuthentication").enabled
+    $auth_windows                     = [bool](Get-WebConfiguration -Location "${site}/${name}" -Filter "system.webserver/security/authentication/windowsAuthentication").enabled
+  }catch{
+    $provider_error = 'Get-WebConfiguration error. Please review the identity accessing physicalpath.'
+  }
 
```

Comment:
By default `Get-WebConfiguration -Location` reads a web.config file in the application's physicalPath, which may not always be available if `physicalPath` is a UNC path. 

Rather than outputting a PowerShell error, this change catches the PowerShell error and returns a Puppet error to the user (In this case, via a private `_provider_error` key) and report that some properties could not be read and setting those properties to 'unknown'.

I needed this while testing, and could remove it. 

But we since we should want to minimize potential failures in PowerShell, and we sanitize input that we pass to PowerShell, we could consider catching and sanitizing output from PowerShell, and pass that output back to Puppet ... wherever we call a PowerShell command.